diff --git a/jbpm-services/jbpm-services-cdi/pom.xml b/jbpm-services/jbpm-services-cdi/pom.xml
index 42c91d4..2652625 100644
--- a/jbpm-services/jbpm-services-cdi/pom.xml
+++ b/jbpm-services/jbpm-services-cdi/pom.xml
@@ -6,7 +6,7 @@
     <artifactId>jbpm-services</artifactId>
     <version>6.5.0.Final-redhat-14</version>
   </parent>
-  <artifactId>jbpm-services-cdi</artifactId>
+  <artifactId>jbpm-services-cdi-DEBUG</artifactId>
   <name>jBPM :: CDI Services</name>
   
   <dependencies>
diff --git a/jbpm-services/jbpm-services-cdi/src/main/java/org/jbpm/services/cdi/impl/manager/InjectableRegisterableItemsFactory.java b/jbpm-services/jbpm-services-cdi/src/main/java/org/jbpm/services/cdi/impl/manager/InjectableRegisterableItemsFactory.java
index 65f9451..4b85e56 100644
--- a/jbpm-services/jbpm-services-cdi/src/main/java/org/jbpm/services/cdi/impl/manager/InjectableRegisterableItemsFactory.java
+++ b/jbpm-services/jbpm-services-cdi/src/main/java/org/jbpm/services/cdi/impl/manager/InjectableRegisterableItemsFactory.java
@@ -165,9 +165,12 @@ public class InjectableRegisterableItemsFactory extends DefaultRegisterableItems
             }
         }
         try {
+        	int i = 0;
             for (WorkItemHandlerProducer producer : workItemHandlerProducer) {
+            	i++;
                 handler.putAll(producer.getWorkItemHandlers(manager.getIdentifier(), parameters));
             }
+            System.out.println("DEBUG: " + manager.getIdentifier() + " has " + i + " workItemHandlerProducer");
         } catch (Exception e) {
             // do nothing as work item handler is considered optional
             logger.warn("Exception while evalutating work item handler prodcuers {}", e.getMessage());
